src/simplewallet/simplewallet.cpp

```diff
@@ -6773,7 +6773,14 @@ bool simple_wallet::get_tx_key(const std::vector<std::string> &args_)
 
   crypto::secret_key tx_key;
   std::vector<crypto::secret_key> additional_tx_keys;
-  if (m_wallet->get_tx_key(txid, tx_key, additional_tx_keys))
+
+  bool found_tx_key = m_wallet->get_tx_key(txid, tx_key, additional_tx_keys);
+  if (!found_tx_key)
+  {
+    found_tx_key = m_wallet->get_tx_key_device(txid, tx_key, additional_tx_keys);
+  }
```

Comment:
Does this need to be one in the caller ?

Reply:
I wanted to discuss this. How to do it correctly. 
`get_tx_key` is const so I cannot just fallback to device in the `wallet::get_tx_key` if it needs to be loaded from the device.

Reply:
Sounds like the device fields should be mutable then. Good old C++ bodge ? :)

Reply:
Soo do you prefer `wallet2::get_tx_key` is not `const` anymore and does the device call if needed? I would like it more. 

Or I can rename `get_tx_key` to `get_tx_key_stored` which would be const, the `get_tx_key`  would do `get_tx_key_stored` first and then query the device. Does that sound OK?

Reply:
It does sound OK, thanks. Though "_stored" is not very enlightening as to what it does, but it'll do.

Reply:
Yeah, I sometimes struggle with names, maybe cached? :)